### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by BertrandBorel** [zauberzeug/nicegui#1890] [Ideas / Feature Requests]
### Question

I want to add pagination to my page, the following code works :

```py
with ui.element("div").classes("flex justify-center w-full")  :       
    a = ui.element("q-pagination").props("min=1 max=10 :model-value=1 direction-links").on("update:model-value", lambda x: a.props(f":model-value={x.args}"))
```

I think the code could be improved and cleaner but I don't know how to do this.
I searched in the Nicegui documentation but couldn't find anything.
If you have any advice I'm interested !

**Comment by falkoschindler** [comment]
That's a great start, @BertrandBorel!

I quickly transformed it into a `Pagination` class and extended it with some already existing mixins:

```py
from nicegui.elements.mixins.disableable_element import DisableableElement
from nicegui.elements.mixins.value_element import ValueElement

class Pagination(ValueElement, DisableableElement):

    def __init__(self, min: int, max: int, *, direction_links: bool = False,
                 value: Optional[int] = ..., on_change: Optional[Callable[..., Any]] = None) -> None:
        super().__init__(tag='q-pagination', value=value, on_value_change=on_change)
        self._props['min'] = min
        self._props['max'] = max
        self._props['direction-links'] = direction_links
        if value is ...:
            self.value = min

Pagination(1, 10, direction_links=True)
```

This way the `value` is bindable and you can enable/disable the element. What's missing is an integration into ui.py, some documentation and a pytest. Anyone up for a pull request? 🙂

**Reply by adarshpunj** [reply]
@falkoschindler done, please check #1899

**Reply by BertrandBorel** [reply]
@falkoschindler Thanks a lot !

**Reply by falkoschindler** [reply]
That was quick. Thanks @adarshpunj! 😀

**Reply by falkoschindler** [reply]
PR #1899 has been released in version 1.4.2.
